File: LeetCode/481-magical-string/ex_481.py
class Solution(object):
    def magicalString(self, n):
        """
        :type n: int
        :rtype: int
        """
        s = [1, 2, 2]
        i = 2
        while len(s) < n:
            s += s[i] * [(3 - s[-1])]
            i += 1
        return s[:n].count(1)

File: LeetCode/481-magical-string/ex_481.cpp
class Solution {
public:
    int magicalString(int n) {
        vector<int> s{1, 2, 2};
        int i = 2;
        while (s.size() < n) s.insert(s.end(), s[i++], 3 - s.back());
        return count(s.begin(), s.begin() + n, 1);
    }
};